File: Src/Icm.Core/Timers.vb
Namespace Icm.Timers

    Public Class Timer
        Inherits System.Timers.Timer

        Public Sub New()

        End Sub

        Public Sub New(ByVal interval As Double)
            MyBase.New(interval)
        End Sub

        Property NextElapsed() As Date
            Get
                Return Now.Add(New TimeSpan(0, 0, 0, 0, CInt(Interval)))
            End Get
            Set(ByVal Value As Date)
                Interval = Value.Subtract(Now).TotalMilliseconds
            End Set
        End Property
    End Class

End Namespace

File: Src/Icm.Core/Timers.cs
using System;

namespace Icm.Timers
{

	public class Timer : System.Timers.Timer
	{


		public Timer()
		{
		}

		public Timer(double interval) : base(interval)
		{
		}

		public System.DateTime NextElapsed {
			get { return DateTime.Now.Add(new TimeSpan(0, 0, 0, 0, Convert.ToInt32(Interval))); }
			set { Interval = value.Subtract(DateTime.Now).TotalMilliseconds; }
		}
	}

}
